lib/pf/Switch/Cisco/Catalyst_2960.pm

```diff
@@ -513,9 +513,10 @@ sub returnRadiusAccessAccept {
     if ( isenabled($self->{_UrlMap}) && $self->externalPortalEnforcement ) {
         if( defined($args->{'user_role'}) && $args->{'user_role'} ne "" && defined($self->getUrlByName($args->{'user_role'}))){
             my $mac = $args->{'mac'};
+            $args->{'session_id'} = "sid".$self->setSession($args);
             my $redirect_url = $self->getUrlByName($args->{'user_role'});
-            $args->{'session_id'} = "sid".$self->setSession($args) if ($redirect_url =~ /\$session_id/);
-            $redirect_url =~ s/\$([a-zA-Z_0-9]+)/$args->{$1} \/\/ ''/ge;
+            $redirect_url .= '/' unless $redirect_url =~ m(\/$);
+            $redirect_url .= $args->{'session_id'};
```

Comment:
You're right but I don't see the point that much. A normal flow would want the portal to be displayed with some instructions with a link to http://www.antivirus.ca

Redirecting the user straight to that website wouldn't make much sense.

IMO general configuration shouldn't be built to handle the 0.01% corner cases unless it has absolutely no impact on the interaction.

Also, if someone would want to handle that extreme corner case, he could do so using the RADIUS filters.